openqaoa/optimizers/training_vqa.py

```diff
@@ -212,13 +218,17 @@ def optimize_this(self, x):
         if self.save_to_csv:
             save_parameter('param_log', deepcopy(x))
         
-        callback_cost = self.vqa.expectation(self.variational_params)
-        
+        callback_cost = self.vqa.expectation(self.variational_params)        
         log_dict.update({'cost': callback_cost})
+
         current_eval = self.log.func_evals.best[0]
         current_eval += 1
         log_dict.update({'func_evals': current_eval})
-        
+
+        eval_number = self.log.eval_number.history[-1] if len(self.log.eval_number.history) > 0 else 0
+        eval_number += 1
+        log_dict.update({'eval_number': eval_number})  #this one will say which evaluation is the optimized one  
```

Comment:
@vishal-ph in light of what we were discussing yesterday, could you check this?

Reply:
If the goal here is to just record the evluation number where the cost is the lowest.
Then all we need to do here is
```
log_dict.update({'eval_number': current_eval})
```

`history_update_bool` for `eval_number` can be `False`

If the cost is not updated, `eval_number` does not get updated, even if a value is passed in,.